### PR TITLE
[WIP] Fixes #11215

### LINKED ISSUE #11215: Error in \Phalcon\Text::dynamic()
Please replace on https://github.com/phalcon/cphalcon/blame/master/phalcon/text.zep#L272
`let words = explode("|", matches[1]);`
to 
`let words = explode(separator, matches[1]);`

### PR DESCRIPTION
**Pull request by Green-Cat**

### CODE REVIEW

File: phalcon/text.zep [hunk 1]
```diff
@@ -269,7 +269,7 @@ abstract class Text
 		while memstr(result, leftDelimiter) {
 			let result = preg_replace_callback(pattern, function (matches) {
 				var words;
-				let words = explode("|", matches[1]);
+				let words = explode(separator, matches[1]);
```
Comment (sergeyklay) [inline phalcon/text.zep]: This is the anonymous function. Use:
```zephir
function (matches) use (separator) {
   ...
}
```

Comment (Green-Cat) [inline phalcon/text.zep]: Oops, totally missed the anonymous function. Thanks!

**Comment by sergeyklay** [comment]
@Green-Cat Could you please deal with this?

**Comment by Green-Cat** [comment]
@sergeyklay Any idea how to go about fixing it? Zephir doesn't support use () in closures.

**Comment by sergeyklay** [comment]
For example get rid form the `preg_replace_callback`

File: phalcon/text.zep [hunk 2]
```diff
@@ -265,17 +267,24 @@ abstract class Text
...
+			let result = preg_replace_callback(pattern, "self::dynamic_replace_callback", result);
 		}
 
 		return result;
 	}
+	
+	/**
+	 * The callback function used in Phalcon\Text::dynamic to select a random text
+	 */
+	protected function dynamic_replace_callback(matches)
```
Comment (sergeyklay) [inline phalcon/text.zep]: mb static?

**Comment by sergeyklay** [comment]
Fixed in the `2.1.x` branch.

**Pull request closed without merging**